src/together/lib/cli/api/utils.py

```diff
@@ -40,11 +48,100 @@ def convert(
             return bool(value)
         except ValueError:
             self.fail(
-                _("{value!r} is not a valid {type}.").format(value=value, type=self.name),
+                _("{value!r} is not a valid {type}.").format(
+                    value=value, type=self.name
+                ),
                 param,
                 ctx,
             )
 
 
 INT_WITH_MAX = AutoIntParamType()
 BOOL_WITH_AUTO = BooleanWithAutoParamType()
+
+
+def _human_readable_time(timedelta: float) -> str:
+    """Convert a timedelta to a compact human-readble string
+    Examples:
+        00:00:10 -> 10s
+        01:23:45 -> 1h 23min 45s
+        1 Month 23 days 04:56:07 -> 1month 23d 4h 56min 7s
+    Args:
+        timedelta (float): The timedelta in seconds to convert.
+    Returns:
+        A string representing the timedelta in a human-readable format.
+    """
+    units = [
+        (30 * 24 * 60 * 60, "month"),  # 30 days
+        (24 * 60 * 60, "d"),
+        (60 * 60, "h"),
+        (60, "min"),
+        (1, "s"),
+    ]
+
+    total_seconds = int(timedelta)
+    parts: List[str] = []
+
+    for unit_seconds, unit_name in units:
+        if total_seconds >= unit_seconds:
+            value = total_seconds // unit_seconds
+            total_seconds %= unit_seconds
+            parts.append(f"{value}{unit_name}")
+
+    return " ".join(parts) if parts else "0s"
+
+
+def generate_progress_bar(
+    finetune_job: Union[Data, FinetuneResponse, _FinetuneResponse], current_time: datetime, use_rich: bool = False
+) -> str:
+    """Generate a progress bar for a finetune job.
+    Args:
+        finetune_job: The finetune job to generate a progress bar for.
+        current_time: The current time.
+        use_rich: Whether to use rich formatting.
+    Returns:
+        A string representing the progress bar.
+    """
+    progress = "Progress: [bold red]unavailable[/bold red]"
+    if finetune_job.status in COMPLETED_STATUSES:
+        progress = "Progress: [bold green]completed[/bold green]"
+    elif finetune_job.updated_at is not None:
+        updated_at_unknown_type = finetune_job.updated_at
+        if isinstance(updated_at_unknown_type, str):
+            # TODO: Retrieve Fine-tuning job function returns string instead of datetime
```

Comment:
put up a fix for this: https://github.com/togethercomputer/openapi/pull/181

as a side note - do you know why we have a FinetuneResponse and a FinetuneResponseTruncated model in the openapi spec? They both seem relatively non-truncated lol